Plugin does not seem to work when you don't have Gradle installed

**Describe the bug**
All our projects are using gradle wrapper, so Gradle itself is not installed to my system and its not in my path. Does it mean that the plugin does not work with Gradle wrapper? Connecting to Xray works and CI tab works as well. Its the local tab thats failing.

**Versions**
- IntelliJ IDEA 2021.1.3 Ultimate Edition
- JFrog IDEA plugin version: 1.9.1
- Operating system: MacOS 10.15.7
- Xray version: 3.27.2

**Additional context**
```
20:44	JFrog Xray scan failed: [ERROR] Could not scan Gradle project dependencies, because Gradle CLI is not in the PATH.

20:44	JFrog Xray scan failed
			[ERROR] 
			java.lang.NullPointerException
			at com.jfrog.ide.common.scan.ScanManagerBase.extractComponents(ScanManagerBase.java:132)
			at com.jfrog.ide.common.scan.ScanManagerBase.scanAndCacheArtifacts(ScanManagerBase.java:150)
			at com.jfrog.ide.idea.scan.ScanManager.scanAndUpdate(ScanManager.java:108)
			at com.jfrog.ide.idea.scan.ScanManager$1.run(ScanManager.java:145)
			at com.intellij.openapi.progress.impl.CoreProgressManager$TaskRunnable.run(CoreProgressManager.java:998)
			at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcessWithProgressAsync$5(CoreProgressManager.java:497)
			at com.intellij.openapi.progress.impl.ProgressRunner.lambda$submit$3(ProgressRunner.java:228)
			at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:178)
			at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:688)
			at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgr... (show balloon)

```

--- a/src/main/java/com/jfrog/ide/idea/log/Logger.java
+++ b/src/main/java/com/jfrog/ide/idea/log/Logger.java
@@ -2,6 +2,9 @@ package com.jfrog.ide.idea.log;
 
 import com.intellij.notification.*;
 import com.intellij.openapi.components.ServiceManager;
+import com.intellij.openapi.options.Configurable;
+import com.intellij.openapi.options.ShowSettingsUtil;
+import com.intellij.openapi.project.Project;
 import org.apache.commons.lang3.StringUtils;
 import org.codehaus.plexus.util.ExceptionUtils;
 import org.jfrog.build.api.util.Log;
@@ -103,5 +106,10 @@ public class Logger implements Log {
                 return "[ERROR] " + message;
         }
         return "[INFO] " + message;
+    }
+
+    public static void openSettings(String details, Project project, Class<? extends Configurable> configurable) {
+        EVENT_LOG_NOTIFIER.createNotification(INFORMATION_TITLE, prependPrefix(details, NotificationType.INFORMATION), NotificationType.INFORMATION,
+                ((notification, event) -> ShowSettingsUtil.getInstance().showSettingsDialog(project, configurable))).notify(project);
     }
 }

--- a/src/main/java/com/jfrog/ide/idea/log/Logger.java
+++ b/src/main/java/com/jfrog/ide/idea/log/Logger.java
@@ -108,7 +108,17 @@ public class Logger implements Log {
         return "[INFO] " + message;
     }
 
-    public static void openSettings(String details, Project project, Class<? extends Configurable> configurable) {
+    /**
+     * Add a log message with an open settings link.
+     * Usage example:
+     * Logger.openSettings("It looks like Gradle home was not properly set in your project.
+     * Click <a href=\"#settings\">here</a> to set Gradle home.", project, GradleConfigurable.class);
+     *
+     * @param details      - The log message
+     * @param project      - IDEA project
+     * @param configurable - IDEA settings to open
+     */
+    public static void addOpenSettingsLink(String details, Project project, Class<? extends Configurable> configurable) {
         EVENT_LOG_NOTIFIER.createNotification(INFORMATION_TITLE, prependPrefix(details, NotificationType.INFORMATION), NotificationType.INFORMATION,
                 ((notification, event) -> ShowSettingsUtil.getInstance().showSettingsDialog(project, configurable))).notify(project);
     }

--- a/src/main/java/com/jfrog/ide/idea/scan/GradleScanManager.java
+++ b/src/main/java/com/jfrog/ide/idea/scan/GradleScanManager.java
@@ -77,10 +77,10 @@ public class GradleScanManager extends ScanManager {
     }
 
     /**
-     * Extracts the chosen Gradle executable path from the Gradle plugin. If Gradle is not installed, throw an exception.
+     * Extract the chosen Gradle executable path from the Gradle plugin. If Gradle is not configured well, return null.
      *
-     * @param env - The environment variables to set the JAVA_HOME
-     * @return the chosen Gradle executable path
+     * @param env - The environment variables map to set the JAVA_HOME
+     * @return the chosen Gradle executable path or null
      */
     private String getGradleExeAndJdk(Map<String, String> env) {
         File gradleHome = resolveGradleAndSetJavaHome(env);
@@ -118,7 +118,7 @@ public class GradleScanManager extends ScanManager {
         }
         // Gradle wasn't set properly
         if (isMisconfigurationError(projectSettings.getExternalProjectPath())) {
-            Logger.openSettings("It looks like Gradle home was not properly set in your project. " +
+            Logger.addOpenSettingsLink("It looks like Gradle home was not properly set in your project. " +
                     "Click <a href=\"#settings\">here</a> to set Gradle home.", project, GradleConfigurable.class);
         } else {
             getLog().warn("Can't run Gradle from Gradle settings. Hint - try to reload Gradle project and then refresh the scan.");

--- a/src/main/java/com/jfrog/ide/idea/scan/GradleScanManager.java
+++ b/src/main/java/com/jfrog/ide/idea/scan/GradleScanManager.java
@@ -1,6 +1,7 @@
 package com.jfrog.ide.idea.scan;
 
 import com.intellij.codeInspection.LocalInspectionTool;
+import com.intellij.openapi.components.ServiceManager;
 import com.intellij.openapi.project.Project;
 import com.intellij.openapi.vfs.LocalFileSystem;
 import com.intellij.openapi.vfs.VirtualFile;
@@ -11,9 +12,17 @@ import com.jfrog.ide.common.gradle.GradleTreeBuilder;
 import com.jfrog.ide.common.scan.ComponentPrefix;
 import com.jfrog.ide.idea.inspections.GradleGroovyInspection;
 import com.jfrog.ide.idea.inspections.GradleKotlinInspection;
+import com.jfrog.ide.idea.log.Logger;
 import com.jfrog.ide.idea.ui.ComponentsTree;
 import com.jfrog.ide.idea.ui.filters.filtermanager.ConsistentFilterManager;
+import org.apache.commons.lang3.SystemUtils;
+import org.jetbrains.plugins.gradle.service.GradleInstallationManager;
+import org.jetbrains.plugins.gradle.service.settings.GradleConfigurable;
+import org.jetbrains.plugins.gradle.settings.DistributionType;
+import org.jetbrains.plugins.gradle.settings.GradleProjectSettings;
+import org.jetbrains.plugins.gradle.settings.GradleSettings;
 
+import java.io.File;
 import java.io.IOException;
 import java.nio.file.Path;
 import java.nio.file.Paths;
@@ -34,7 +43,7 @@ public class GradleScanManager extends ScanManager {
     GradleScanManager(Project project, String basePath) throws IOException {
         super(project, basePath, ComponentPrefix.GAV);
         getLog().info("Found Gradle project: " + getProjectName());
-        gradleTreeBuilder = new GradleTreeBuilder(Paths.get(basePath), EnvironmentUtil.getEnvironmentMap());
+        gradleTreeBuilder = new GradleTreeBuilder(Paths.get(basePath), EnvironmentUtil.getEnvironmentMap(), getGradleExe());
     }
 
     @Override
@@ -61,5 +70,53 @@ public class GradleScanManager extends ScanManager {
     @Override
     protected void buildTree(boolean shouldToast) throws IOException {
         setScanResults(gradleTreeBuilder.buildTree(getLog()));
+    }
+
+    /**
+     * Extracts the chosen Gradle executable path from the Gradle plugin. If Gradle is not installed, throw an exception.
+     *
+     * @return the chosen Gradle executable path
+     */
+    private String getGradleExe() {
+        File gradleHome = resolveGradleHome();
+        if (gradleHome == null) {
+            getLog().info("Using Gradle from system path.");
+            return null;
+        }
+        String gradleExe = gradleHome.toPath().resolve("bin").resolve(SystemUtils.IS_OS_WINDOWS ? "gradle.bat" : "gradle").toString();
+        getLog().info("Using Gradle executable " + gradleExe);
+        return gradleExe;
+    }
+
+    private File resolveGradleHome() {
+        GradleProjectSettings projectSettings = GradleSettings.getInstance(project).getLinkedProjectsSettings().stream().findAny().orElse(null);
+        if (projectSettings == null) {
+            getLog().error("Couldn't retrieve Gradle project settings. Hint - make sure the Gradle project was properly imported.");
+            return null;
+        }
+        GradleInstallationManager gradleInstallationManager = ServiceManager.getService(GradleInstallationManager.class);
+        File gradleHome = gradleInstallationManager.getGradleHome(project, projectSettings.getExternalProjectPath());
+        if (gradleHome != null) {
+            return gradleHome;
+        }
+        if (isMisconfigurationError(projectSettings.getExternalProjectPath())) {
+            Logger.openSettings("It looks like Gradle home was not properly set in your project. " +
+                    "Click <a href=\"#settings\">here</a> to set Gradle home.", project, GradleConfigurable.class);
+        } else {
+            getLog().warn("Can't run Gradle from Gradle settings. Hint - try to reload Gradle project and then refresh the scan.");
+        }
+        return null;
+    }
+
+    private boolean isMisconfigurationError(String linkedProjectPath) {
+        GradleProjectSettings projectSettings = GradleSettings.getInstance(project).getLinkedProjectSettings(linkedProjectPath);
+        if (projectSettings != null) {
+            DistributionType distributionType = projectSettings.getDistributionType();
+            // Distribution type has not been chosen or the distribution type is not Gradle wrapper.
+            // If the distribution type is wrapped, it is probable that the Gradle wrapper is not yet created.
+            return distributionType == null || !distributionType.isWrapped();
+        }
+        // Gradle project settings are not set
+        return true;
     }
 }

--- a/src/main/java/com/jfrog/ide/idea/scan/GradleScanManager.java
+++ b/src/main/java/com/jfrog/ide/idea/scan/GradleScanManager.java
@@ -1,5 +1,6 @@
 package com.jfrog.ide.idea.scan;
 
+import com.google.common.collect.Maps;
 import com.intellij.codeInspection.LocalInspectionTool;
 import com.intellij.openapi.components.ServiceManager;
 import com.intellij.openapi.project.Project;
@@ -15,6 +16,7 @@ import com.jfrog.ide.idea.inspections.GradleKotlinInspection;
 import com.jfrog.ide.idea.log.Logger;
 import com.jfrog.ide.idea.ui.ComponentsTree;
 import com.jfrog.ide.idea.ui.filters.filtermanager.ConsistentFilterManager;
+import org.apache.commons.lang3.StringUtils;
 import org.apache.commons.lang3.SystemUtils;
 import org.jetbrains.plugins.gradle.service.GradleInstallationManager;
 import org.jetbrains.plugins.gradle.service.settings.GradleConfigurable;
@@ -26,6 +28,7 @@ import java.io.File;
 import java.io.IOException;
 import java.nio.file.Path;
 import java.nio.file.Paths;
+import java.util.Map;
 
 /**
  * Created by Yahav Itzhak on 9 Nov 2017.
@@ -43,7 +46,8 @@ public class GradleScanManager extends ScanManager {
     GradleScanManager(Project project, String basePath) throws IOException {
         super(project, basePath, ComponentPrefix.GAV);
         getLog().info("Found Gradle project: " + getProjectName());
-        gradleTreeBuilder = new GradleTreeBuilder(Paths.get(basePath), EnvironmentUtil.getEnvironmentMap(), getGradleExe());
+        Map<String, String> env = Maps.newHashMap(EnvironmentUtil.getEnvironmentMap());
+        gradleTreeBuilder = new GradleTreeBuilder(Paths.get(basePath), env, getGradleExeAndJdk(env));
     }
 
     @Override
@@ -75,10 +79,11 @@ public class GradleScanManager extends ScanManager {
     /**
      * Extracts the chosen Gradle executable path from the Gradle plugin. If Gradle is not installed, throw an exception.
      *
+     * @param env - The environment variables to set the JAVA_HOME
      * @return the chosen Gradle executable path
      */
-    private String getGradleExe() {
-        File gradleHome = resolveGradleHome();
+    private String getGradleExeAndJdk(Map<String, String> env) {
+        File gradleHome = resolveGradleAndSetJavaHome(env);
         if (gradleHome == null) {
             getLog().info("Using Gradle from system path.");
             return null;
@@ -88,7 +93,13 @@ public class GradleScanManager extends ScanManager {
         return gradleExe;
     }
 
-    private File resolveGradleHome() {
+    /**
+     * Resolve Gradle executable and Java home from Gradle settings.
+     *
+     * @param env - The environment variables to set the JAVA_HOME
+     * @return gradle executable
+     */
+    private File resolveGradleAndSetJavaHome(Map<String, String> env) {
         GradleProjectSettings projectSettings = GradleSettings.getInstance(project).getLinkedProjectsSettings().stream().findAny().orElse(null);
         if (projectSettings == null) {
             getLog().error("Couldn't retrieve Gradle project settings. Hint - make sure the Gradle project was properly imported.");
@@ -97,8 +108,15 @@ public class GradleScanManager extends ScanManager {
         GradleInstallationManager gradleInstallationManager = ServiceManager.getService(GradleInstallationManager.class);
         File gradleHome = gradleInstallationManager.getGradleHome(project, projectSettings.getExternalProjectPath());
         if (gradleHome != null) {
+            // Gradle seems to set properly, try to set JAVA_HOME environment variable.
+            String javaHome = gradleInstallationManager.getGradleJvmPath(project, projectSettings.getExternalProjectPath());
+            if (StringUtils.isNotBlank(javaHome)) {
+                getLog().info("Using Java home: " + javaHome);
+                env.put("JAVA_HOME", javaHome);
+            }
             return gradleHome;
         }
+        // Gradle wasn't set properly
         if (isMisconfigurationError(projectSettings.getExternalProjectPath())) {
             Logger.openSettings("It looks like Gradle home was not properly set in your project. " +
                     "Click <a href=\"#settings\">here</a> to set Gradle home.", project, GradleConfigurable.class);

